### DISCUSSION TITLE
Application breaks when "undefined" is stored with Alpine.$persist()

### DISCUSSION THREAD

**Discussion by simonferndriger** [alpinejs/alpine#4109] [7. Bugs (archived)]
It can happen that `undefined` gets stored as a value for a key in localstorage via `Alpine.$persist()` by accident. In that case, the app breaks and is unusable!

This is a major issue and should be fixed IMHO.

---

**Example:** https://codepen.io/simonferndriger/pen/QWZrLrX
_Uncaught SyntaxError: "undefined" is not valid JSON_

**Comment by SimoTod** [comment]
FYI, There was already raised and a PR was merged not long ago: https://github.com/alpinejs/alpine/pull/4091

**Comment by simonferndriger** [comment]
Thanks. Sorry for the duplicate in that case.

**Reply by SimoTod** [reply]
No worries. I don't think the fix has been released yet 👍
